### PR TITLE
docs: Add breaking changes section to reference docs

### PR DESCRIPTION
**Pull request by catmsred**
## Proposed Commit Message
```
docs: Add breaking changes section to reference
```

## Additional Context
In order to have snapd seeding changes to reference from the noble release docs

## Checklist

- [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [ ] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [x] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: doc/rtd/reference/breaking_changes.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.. _breaking_changes:
+
+Breaking Changes
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:3]:
```suggestion
Breaking changes
```

Comment (s-makin) [reply doc/rtd/reference/breaking_changes.rst:3]: (Headings are always in sentence case)

File: doc/rtd/reference/breaking_changes.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+runcmd:
+    - [ snap, wait, system, seed.loaded ]
+    - [ snap, install, mc-installer ]
+
+
+23.4 - status codes
+===================
+
+something something if you have a script that relies on cloud-init status
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:35]: Something something? :D

File: doc/rtd/reference/breaking_changes.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Breaking Changes
+****************
+
+This section provides reference and guidance on specific breaking changes to 
+cloud-init releases. 
+
+24.1 - removed Ubuntu's ordering dependency on snapd.seeded
+===========================================================
+
+In Ubuntu releases, cloud-init will no longer wait on snapd preseeding to run.
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:12]:
```suggestion
In Ubuntu releases, cloud-init will no longer wait on ``snapd`` pre-seeding to run.
```

File: doc/rtd/reference/breaking_changes.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+If a user-provided script relies on a snap, it must now be prefixed with
+`snap wait system seed.loaded` to ensure the snaps are ready for use.  For
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:14]:
```suggestion
``snap wait system seed.loaded`` to ensure the snaps are ready for use. For
```

File: doc/rtd/reference/breaking_changes.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+example, a cloud config that previously included:
+
+.. code-block:: yaml
+
+runcmd:
+    - [ snap, install, mc-installer ]
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:20]:
```suggestion
.. code-block:: yaml

   runcmd:
       - [ snap, install, mc-installer ]
```

File: doc/rtd/reference/breaking_changes.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+Will now need to be:
+
+.. code-block:: yaml
+
+runcmd:
+    - [ snap, wait, system, seed.loaded ]
+    - [ snap, install, mc-installer ]
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:29]:
```suggestion
.. code-block:: yaml

   runcmd:
       - [ snap, wait, system, seed.loaded ]
       - [ snap, install, mc-installer ]
```

File: doc/rtd/reference/breaking_changes.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
+.. _breaking_changes:
+
+Breaking changes
+****************
+
+This section provides reference and guidance on specific breaking changes to
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:6]:
```suggestion
This section provides guidance on specific breaking changes to 
```

File: doc/rtd/reference/breaking_changes.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      - [ snap, wait, system, seed.loaded ]
+      - [ snap, install, mc-installer ]
+
+
+23.4 - added status code for recoverable error
+==============================================
+
+Cloud-init return codes have been extended with a new error code (2),
+which will be returned when cloud-init experiences an error that it can
+recover from.  See :ref:`this page which documents the change<error_codes>`.
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:37]:
```suggestion
recover from. See :ref:`this page which documents the change <error_codes>`.
```

File: doc/rtd/reference/breaking_changes.rst [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+
+23.2 - kernel commandline
+=========================
+
+The ds= kernel commandline value is used to forcibly select a specific
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:43]:
```suggestion
The ``ds=`` kernel commandline value is used to forcibly select a specific
```

File: doc/rtd/reference/breaking_changes.rst [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+datasource in cloud-init. Prior to 23.2, this only optionally selected
+the ``NoCloud`` datasource.
+
+Anyone that previously had a matching `ds=nocloud*` in their kernel commandline
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:47]:
```suggestion
Anyone that previously had a matching ``ds=nocloud*`` in their kernel commandline
```

File: doc/rtd/reference/breaking_changes.rst [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+that did not want to use the NoCloud datasource may experience broken behavior
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:48]:
```suggestion
that did not want to use the ``NoCloud`` datasource may experience broken behavior
```

File: doc/rtd/reference/breaking_changes.rst [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+as a result of this change.
+
+Workarounds include updating the kernel commandline and optionally configuring
+a datasource_list in /etc/cloud/cloud.cfg.d/*.cfg.
```
Comment (s-makin) [inline doc/rtd/reference/breaking_changes.rst:52]:
```suggestion
a ``datasource_list`` in ``/etc/cloud/cloud.cfg.d/*.cfg``.
```

**Review by s-makin — COMMENTED** [review]
A few more formatting suggestions to avoid triggering the spellchecker

**Review by s-makin — APPROVED** [review]
LGTM! :) thanks for the changes

**Review by holmanb — APPROVED** [review]
Thanks @catmsred, this looks good to me!

**Pull request merged**